dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/internal/net/TracingRequestInterceptor.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.tracing.internal.net
+
+import com.datadog.android.core.internal.net.RequestInterceptor
+import com.datadog.android.core.internal.utils.devLogger
+import com.datadog.android.tracing.AndroidTracer
+import com.datadog.android.tracing.internal.TracesFeature
+import datadog.trace.api.DDTags
+import datadog.trace.api.interceptor.MutableSpan
+import io.opentracing.Span
+import io.opentracing.Tracer
+import io.opentracing.propagation.Format
+import io.opentracing.propagation.TextMapInject
+import io.opentracing.tag.Tags
+import io.opentracing.util.GlobalTracer
+import java.io.PrintWriter
+import java.io.StringWriter
+import java.util.concurrent.atomic.AtomicReference
+import okhttp3.Request
+import okhttp3.Response
+
+internal class TracingRequestInterceptor : RequestInterceptor {
+
+    private val localTracerReference: AtomicReference<Tracer> = AtomicReference()
+
+    private val startedSpans = mutableMapOf<String, Span>()
```

Comment:
is OkHttp using same thread for all the requests ? If not I am afraid here you are going to have a ConcurrentModificationException at some point. Can you check this by adding some tests ?